src/black/files.py

```diff
@@ -244,6 +256,11 @@ def gen_python_files(
         if child.is_dir():
             # If gitignore is None, gitignore usage is disabled, while a Falsey
             # gitignore is when the directory doesn't have a .gitignore file.
+            if gitignore is None:
+                _gitignore = None
+            else:
+                _gitignore = gitignore.copy()
+                _gitignore[child] = get_gitignore(child)
```

Comment:
I'll also use the path from root (`root / child`) to the corresponding .gitignore. This solves producing additional and incorrect output when running black on a directory 👌 Change also applied on `src/black/__init__.py` (key `p` in the `_gitignore` dictionary was changed to `root / p`)